plugins/thunderbolt/fu-thunderbolt-device.c

```diff
@@ -320,6 +320,78 @@ fu_thunderbolt_device_get_attr_uint16(FuThunderboltDevice *self, const gchar *na
 	return (guint16)val;
 }
 
+static gboolean
+fu_thunderbolt_device_set_port_offline(FuUdevDevice *device)
```

Comment:
you should be passing in the GError **error as an argument here and then using that in this function rather than local errors.  The idea is supposed to be that if you set FALSE, then you need to return something in error.

The way this function is working right now, you're setting all those errors, but you're discarding them as they're local to the function.